### PR TITLE
Issue 786  Add possibility to add record without specify primary key if it has a default value (PostgresSql)

### PR DESCRIPTION
**Pull request by VincenzoCarlino**

### CODE REVIEW

File: RepoDb.Core/RepoDb/Cachers/CommandTextCache.cs [hunk 1]
```diff
@@ -1093,7 +1091,7 @@ private static DbField GetPrimaryField(BaseRequest request)
                         null,
                         null,
                         null,
-                        null);
+                        null);// TODO understand how set thath a primary key can have a default value, and which is the desired behavior during insert operation of entity without explits fields
```
Comment (mikependon) [inline RepoDb.Core/RepoDb/Cachers/CommandTextCache.cs:1094]: Can we remove the inline comment?

File: RepoDb.Core/RepoDb/Cachers/CommandTextCache.cs [hunk 2]
```diff
@@ -1125,7 +1123,7 @@ private static DbField GetIdentityField(BaseRequest request)
                         null,
                         null,
                         null,
-                        null);
+                        null);// TODO understand how set thath an identity can have a default value, and which is the desired behavior during insert operation of entity without explits fields
```
Comment (mikependon) [inline RepoDb.Core/RepoDb/Cachers/CommandTextCache.cs:1126]: Same here.

File: RepoDb.Core/RepoDb/DbField.cs [hunk 3]
```diff
@@ -78,6 +104,11 @@ public class DbField : IEquatable<DbField>
         /// </summary>
         public bool IsNullable { get; }
 
+        /// <summary>
+        /// Gets tha value wheter the column has a default value
```
Comment (mikependon) [inline RepoDb.Core/RepoDb/DbField.cs:108]: Close with . period?

**Comment by mikependon** [comment]
@VincenzoCarlino - I had added you as a collaborator. Can you create a branch for this PR? I would like to test the changes in all solutions of RepoDB including the (SQL Server, PostgreSQL, MySQL, SQLite and the BulkOperations).

**Comment by VincenzoCarlino** [comment]
Done, I've created the branch on the repo:
https://github.com/mikependon/RepoDB/tree/issue/786

**Comment by mikependon** [comment]
Hey mate, first, I am very thankful that you issued this PR, and please continue to do so.

Apology, but it sounds to me that adding the default value to fix the issue of the identity is not the right solution. I think, your problem is actually how the identity is being interpreted by the library. As of writing this, in PostgreSQL, if you use the GENERATED IDENTITY AS ALWAYS as your identity generator, then it will work like a charm. The other method via UUID is not yet supported, but we are on top of it to solve on the next coming release.

On the other hand, the changes you made is one-sided to PosgtreSQL and the other packages (i.e.: MySQL, SQLite and SQLServer) were not covered.

This is just an FYI, it seems that this PR cannot be approved.

In relation to your issue, I am planning to work on it after the upcoming events of mine this week. So hopefully, you can wait for that?

**Comment by VincenzoCarlino** [comment]
@mikependon Yeah sure np, for the moment a build the DLL with my changement so I can still go on, so for the moment is no-blocking for me. Thanks :)

**Pull request closed without merging**